code/modules/admin/sql_notes.dm

```diff
@@ -257,3 +267,53 @@
 		output += ruler
 	usr << browse(output.Join(""), "window=show_notes;size=900x500")
 
+/proc/toggle_note_publicity(note_id)
+	if(!usr)
+		return
+	if(!check_rights(R_ADMIN|R_MOD))
+		return
+	if(!SSdbcore.IsConnected())
+		if(usr)
+			to_chat(usr, "<span class='danger'>Failed to establish database connection.</span>")
+		return
+	if(!note_id)
+		return
+	note_id = text2num(note_id)
+	var/datum/db_query/query_find_note_edit = SSdbcore.NewQuery("SELECT ckey, notetext, adminckey, automated, public FROM notes WHERE id=:note_id", list(
+		"note_id" = note_id
+	))
+	if(!query_find_note_edit.warn_execute())
+		qdel(query_find_note_edit)
+		return
+	if(query_find_note_edit.NextRow())
+		var/target_ckey = query_find_note_edit.item[1]
+		var/text = query_find_note_edit.item[2]
+		var/adminckey = query_find_note_edit.item[3]
+		var/automated = query_find_note_edit.item[4]
+		var/public = query_find_note_edit.item[5]
```

Comment:
We don't need the text or target ckey here surely right?

Reply:
Its used in the logging.